[Teng Wye] iP

nopehax opened the pull request:
# Duke Task Manager
🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧
 **Duke** hasn't reached it's _final form yet_!!! 
🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧
What it can do
* add tasks
* delete tasks
* find tasks
* display list

How to use?
1. download latest release
2. run on your pc

An excerpt:
```java
public ArrayList<Task> searchFor(String keyword) {
        ArrayList<Task> output = new ArrayList<>();
        for (Task t : this.store) {
            if (t.getDescription().contains(keyword)) {
                output.add(t);
            }
        }
        return output;
    }
```

Things to do:
- [x] Update PR description with GFMD elements 
- [ ] Week 4 iP task
- [ ] package new release

> uhhh, i have no idea what to do with blockquotes
>>me too :(
>> count me in

You can see more of my projects [here](github.com/nopehax)!
`System.out.print("Hello, World!")`

Commit:
Add Gradle support

nopehax committed:
added lvl 1 code

nopehax committed:
added lvl 2

nopehax committed:
added new class Task and lvl 3

nopehax committed:
added new classes and lvl 4

nopehax committed:
code cleanup

nopehax committed:
added more responses

nopehax committed:
added new exception classes and lvl 5

nopehax committed:
added lvl 6

nopehax committed:
changed commands array to enum and change to switch statements

nopehax committed:
edited testing files

nopehax committed:
removed unused nullpointerexception

nopehax committed:
removed unused dukenotaskexception

nopehax committed:
add a way to save and load data from file

nopehax committed:
add method where deleting task from list also deletes it from save file

nopehax committed:
Add a way to keep track if task is done in save file

nopehax committed:
Clean up code

nopehax committed:
gitignore save file

nopehax committed:
gitignore save file

nopehax committed:
Update Task classes so that they store date/time as LocalDate/LocalTime

nopehax committed:
Merge branch 'branch-Level-7'

nopehax committed:
Settle conflicts when merging branch 'branch-Level-8'

nopehax committed:
Refactor the code to extract out closely related code as classes

nopehax committed:
Close scanner and add newline at EOF

nopehax committed:
Organize classes into their respective packages

nopehax committed:
Remove empty line in the middle of import statements in Storage.java

nopehax committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

nopehax committed:
Add test for some methods in Duke, Parser and TaskList class

nopehax committed:
Correct mainClassName in build.gradle

nopehax committed:
Add Javadoc to Parser.java

nopehax committed:
Add Javadoc comments

nopehax committed:
Tweak code to comply with Java coding standard (basic)

nopehax committed:
Add find command

nopehax committed:
Merge branch 'branch-A-JavaDoc'

nopehax committed:
Resolve conflict when merging branch 'branch-A-CodingStandard'

nopehax committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Ui.java

nopehax committed:
Edit switch blocks to follow Java coding standards

nopehax committed:
Revert "Edit switch blocks to follow Java coding standards"

This reverts commit 1f8d2f55090a097e49f9f8d09905b039c21e4b73.

nopehax committed:
Revert "Revert "Edit switch blocks to follow Java coding standards""

This reverts commit a0a8549b7abf71eb36bd6727fee8b85b44640a03.

nopehax committed:
Format code to follow Java coding standards

nopehax committed:
Format code to follow Java coding standards

nopehax committed:
Format code to follow Java coding standards

nopehax committed:
Configure Checkstyle and change code to comply with Checkstyle

mqchua commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,165 @@
+            throw new WrongArgumentException(arr[1], e);
+        }
+
+        switch (command) {
```
Is there a reason why 2 switch commands is needed? 
Can it be condensed into a single switch statement? 🤔

mqchua commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,3 @@
+todo something
+deadline eat /by tmr
+event picnic /at 9 Aug
```
I think it will be great to add testing for other commands such as list as well 😄

mqchua reviewed (commented):
Overall great PR! Just some nit picks 👍🏻

mqchua commented on src/test/java/duke/util/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    @Test
+    public void deleteTask() {
```
Will also be great to include additional tests for unmark/mark and Event creation 😄

feliciagan commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,165 @@
+                throw new NoArgumentException(0, e);
+            }
+            break;
+        case deadline:
+            try {
+                String[] desc = arr[1].split(" /by ");
+                list.add(new Deadline(desc[0], desc[1]));
+            } catch (ArrayIndexOutOfBoundsException | DateTimeParseException e) {
+                if (fromSave) {
+                    throw new FileParseException(input, e);
+                } else {
+                    if (e instanceof ArrayIndexOutOfBoundsException) {
+                        throw new NoArgumentException(1, e);
+                    } else {
+                        // e will definitely be a DateTimeParseException
+                        throw new WrongArgumentException(((DateTimeParseException) e).getParsedString(), e);
+                    }
+                }
+            }
+            break;
+        case event:
+            try {
+                String[] desc = arr[1].split(" /at ");
+                list.add(new Event(desc[0], desc[1]));
+            } catch (ArrayIndexOutOfBoundsException | DateTimeParseException e) {
+                if (fromSave) {
+                    throw new FileParseException(input, e);
+                } else {
+                    if (e instanceof ArrayIndexOutOfBoundsException) {
+                        throw new NoArgumentException(2, e);
+                    } else {
+                        //e will definitely be a DateTimeParseException
+                        throw new WrongArgumentException(((DateTimeParseException) e).getParsedString(), e);
+                    }
+                }
+            }
```
Perhaps avoid deep nesting if possible?

feliciagan reviewed (commented):
looks good!

nopehax committed:
Add an empty GUI

nopehax committed:
Add code so that it has the same design as seen in JavaFx Tutorial 3

nopehax committed:
Convert to fxml for JavaFx, following Tutorial part 4

nopehax committed:
Incorporate Duke's logic into the GUI

nopehax committed:
Format code to use GUI instead of CLI, add some cosmetic changes

nopehax committed:
Remove some tests since Duke isn't using CLI anymore

nopehax committed:
Change xmlns to ignore version warnings

nopehax committed:
Display images as a Circle instead of ImageView

nopehax committed:
Clean up code

nopehax committed:
Make profile picture bigger

nopehax committed:
Update mainClassName to duke.Launcher

nopehax committed:
Add Java assert to document important assumptions that should hold

nopehax committed:
Make parseListCommands shorter in Parser.java

The method is too long

Refactored such that code that does similar things are extracted out
into new methods

This is to make the code more readable

nopehax committed:
Merge pull request #2 from nopehax/branch-A-Assertions

Add Java assert to document important assumptions that should hold

nopehax committed:
Merge branch 'master' into branch-A-CodeQuality

nopehax committed:
Merge pull request #3 from nopehax/branch-A-CodeQuality

Make parseListCommands shorter in Parser.java

nopehax committed:
Revert "Make parseListCommands shorter in Parser.java"

nopehax committed:
Merge pull request #4 from nopehax/revert-3-branch-A-CodeQuality

Revert "Make parseListCommands shorter in Parser.java"

nopehax committed:
Merge branch 'master' of https://github.com/nopehax/ip

* 'master' of https://github.com/nopehax/ip:
  Revert "Make parseListCommands shorter in Parser.java"
  Make parseListCommands shorter in Parser.java
  Add Java assert to document important assumptions that should hold

nopehax committed:
Comment

nopehax committed:
Made some edits to see what is the code in the current branch

nopehax committed:
Merge pull request #6 from nopehax/branch-A-CodeQuality

nopehax committed:
Changed Parser.java to what it was meant to be

nopehax committed:
Remove empty line

nopehax committed:
Merge pull request #7 from nopehax/branch-A-CodeQuality

The way it was suppose to be

nopehax committed:
Change code to what it was suppose to be

nopehax committed:
Merge pull request #8 from nopehax/branch-A-CodeQuality

Change code to what it was suppose to be

nopehax committed:
Merge branch 'master' of https://github.com/nopehax/ip

nopehax committed:
Set up github actions custom workflow

nopehax committed:
Add new Task - recurring

to simplify recurring task such that users don't have to key it in every
time

recurring task calculates all the dates that task will happen and saves
it

nopehax committed:
Improve on what Recurring tasks can do

made it so that it'll calculate how many times the task is happening
based on the current date and input date

nopehax committed:
Add remaining command to list the dates a recurring task is happening on

to let the user know all past and future dates the task occurs at

also handle the case where the input index is not a recurring task

nopehax committed:
Merge branch 'branch-Extension'

nopehax committed:
Add remaining command to COMMAND_HELP and refactor it to use streams

nopehax committed:
Fixed bug in Recurring task where the date of next task shown is not
accurate
it shows the date of the next next task instead

fixed by changing the line where startDate is set

nopehax committed:
Merge branch 'branch-Extension'

nopehax committed:
Remove checkstyle part as Java CI has been failing checks

nopehax committed:
Revert "Remove checkstyle part as Java CI has been failing checks"

This reverts commit ff7a4e59ff81d7e7e47ad8f072e236490ebcde79.

nopehax committed:
Edit code such that it follows CheckStyle guide

nopehax committed:
Add JavaDocs comment
this commit is just to add branch-Level-10 into the repo as I added the
GUI code in the master branch itself, instead of creating a new branch

nopehax committed:
Merge branch 'branch-Level-10'

nopehax committed:
Edit code to follow CheckStyle guide

nopehax committed:
Edit code in test cases to follow CheckStyle guide

nopehax committed:
Ui changes - program window is not resizable
Fixed bug in recurring task where daily task cannot be parsed correctly

nopehax committed:
Change Duke's name to Doomba to match it's profile pic

nopehax committed:
Add wallpaper to chat background

nopehax committed:
Add screenshot of ui into docs

nopehax committed:
Update project README.md

nopehax committed:
Remove files related to testing in an attempt to fix Java CI build error

nopehax committed:
Add back test files and config file containing checkstyle folder

It was in .gitignored, which was why Java CI failed on gradlew check as
it couldn't find the file

nopehax committed:
Update project README.md

nopehax committed:
Update JUnit tests

nopehax committed:
Update project README.md

nopehax committed:
Remove unnecessary use of wallpaper.png

nopehax committed:
Remove the correct incorrect line

nopehax committed:
Updata code to follow Java Coding Standards

nopehax committed:
Update README.md

nopehax committed:
Update README.md

nopehax committed:
Update README.md

nopehax committed:
Update project README.md

nopehax committed:
Update README.md

nopehax committed:
Update project README.md

nopehax committed:
Fix QuantumRenderer initialization error for Linus and MacOS

nopehax committed:
Fixed issue where jar file does not work on macos and linux
